redis_info.py

```diff
@@ -202,10 +246,52 @@ def dispatch_value(info, key, type, plugin_instance=None, type_instance=None):
     # write_http plugin. See
     # https://github.com/collectd/collectd/issues/716
     val.meta = {'0': True}
+    val.dispatch()
+
+
+def dispatch_llen_key(key_name, key_value, db_index, plugin_instance):
+    """
+    adds dimensions for key value and db_index for
+    dispatching the key_llen gauage
+    """
+    val = collectd.Values(plugin='redis_info')
+    val.type = 'gauge'
+    val.type_instance = 'key_llen'
+    dimensions = build_dimensions(key_name, int(db_index))
```

Comment:
I would just call `_format_dimensions` with the dict that you create in the `build_dimensions` function below instead of having a separate function, especially since the function name is so generic.